pageTests/api/session.test.js

```diff
@@ -103,6 +106,75 @@ describe("api/session", () => {
           })
         );
       });
+      it("logs a login event when a user has logged in", async () => {
+        const validRequest = {
+          method: "POST",
+          body: {
+            code: "ward_code",
+          },
+          headers: {
+            "x-correlation-id": "correlationId",
+          },
+        };
+        const response = {
+          writeHead: jest.fn(),
+          end: jest.fn(),
+        };
+
+        const verifyWardCodeSpy = jest.fn(async () => ({
+          validWardCode: true,
+          ward: { id: 10, code: "MEOW", trustId: 1 },
+        }));
+
+        const tokenGeneratorSpy = jest.fn(() => "generatedToken");
+
+        const logEventSpy = jest.fn(async () => ({
+          logEventResponse: true,
+        }));
+
+        const container = {
+          getTokenProvider: jest.fn(() => ({
+            generate: tokenGeneratorSpy,
+          })),
+          getVerifyWardCode: () => verifyWardCodeSpy,
+          getVerifyTrustAdminCode: () => () =>
+            jest.fn().mockReturnValue({ validAdminCode: false }),
+          getVerifyAdminCode: () =>
+            jest.fn().mockReturnValue({ validAdminCode: false }),
+          getLogEventGateway: () => logEventSpy,
+        };
+        await session(validRequest, response, { container });
+
+        expect(verifyWardCodeSpy).toHaveBeenCalledWith("ward_code");
+        expect(tokenGeneratorSpy).toHaveBeenCalledWith({
+          wardId: 10,
+          wardCode: "MEOW",
+          trustId: 1,
+          type: "wardStaff",
+        });
+        expect(response.writeHead).toHaveBeenCalledWith(
+          201,
+          expect.objectContaining({
+            "Set-Cookie": [
+              expect.stringContaining("generatedToken"),
+              expect.stringContaining("sessionId"),
+            ],
+          })
+        );
```

Comment:
do we need to be asserting on these spies as part of this test - are they not checked as part of other, existing tests?